core/frontend/src/Viewport.ts

```diff
@@ -3556,6 +3557,15 @@ export class ScreenViewport extends Viewport {
       this._webglCanvas = webglCanvas;
 
       this.addChildDiv(this.vpDiv, webglCanvas, 5);
+      // if (this._webglCanvas.width !== this.canvas.width)
+      //   this._webglCanvas.width = this.canvas.width;
+      // if (this._webglCanvas.height !== this.canvas.height)
+      //   this._webglCanvas.height = this.canvas.height;
+      if (this._webglCanvas.width !== this.canvas.width || this._webglCanvas.height !== this.canvas.height) {
+        this._webglCanvas.width = this.canvas.width;
+        this._webglCanvas.height = this.canvas.height;
+        resized2 = true;
+      }
```

Comment:
@pmconne I first tried testing the canvas size inside of updateDimensions (called from updateViewRect), by adding to the test that was there.  That fixed the "scaling" but introduced some black view cases.  So I removed that and put the test here and still fixed the scaling and got different results of black view (better). 

That is when I investigated further and found that there are 4 different behavior cases (see PR description).  So I tried different things, since it wasn't making sense to me:

Fix scale of canvas width/height:
 - Delete larger clone -> original is fine
 - Delete smaller clone -> original is fine
 - Delete smaller original -> clone turns black
 - Delete larger original -> clone turns black sometimes, otherwise clone is fine

Fix scale of canvas width/height and test for resized2 also:
 - Delete larger clone -> original turns black
 - Delete smaller clone -> original is fine
 - Delete smaller original -> clone turns black sometimes, is fine others
 - Delete larger original -> clone turns black
 - Test passes

Fix scale of canvas width/height and test (resized && !resized2):
 - Delete larger clone -> original is fine
 - Delete smaller clone -> original is fine
 - Delete smaller original -> clone turns black
 - Delete larger original -> clone is fine
 - Test passes
	
Comment out this.target.onResized() - (disposes fbo):
 - Delete larger clone -> original is fine
 - Delete smaller clone -> original is fine
 - Delete smaller original -> clone is fine
 - Delete larger original -> clone is fine
 - Test fails: 1) ViewManager
 - - should resize fbo properly after dropping a recently-resized viewport:  AssertionError: expected false to be true
tried several combinations of testing

Commenting out onResized (below) makes no sense to me why that would fix the black view, but this is where I am at currently, if you have any ideas, and why I just made this a draft.  (I also don't know why some of the black view cases are  not consistent.)

Reply:
Resolved per our discussion.